modules/social_features/social_album/social_album.module

```diff
@@ -0,0 +1,351 @@
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Core\Url;
+use Drupal\social_album\Form\SocialAlbumImageForm;
+use Drupal\social_album\Form\SocialAlbumPostForm;
+
+/**
+ * Implements hook_theme().
+ */
+function social_album_theme($existing, $type, $theme, $path) {
+  return [
+    'social_album_post' => [
+      'variables' => [
+        'items' => NULL,
+        'album' => NULL,
+      ],
+    ],
+  ];
+}
+
+/**
+ * Implements hook_entity_extra_field_info().
+ */
+function social_album_entity_extra_field_info() {
+  return [
+    'node' => [
+      'album' => [
+        'display' => [
+          'social_album_images' => [
+            'label' => t('Images'),
+            'visible' => FALSE,
+          ],
+        ],
+      ],
+    ],
+  ];
+}
+
+/**
+ * Implements hook_entity_view().
+ */
+function social_album_entity_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
+  if ($display->getComponent('social_album_images')) {
+    $build['social_album_images'] = views_embed_view('albums', 'embed_album_overview');
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function social_album_preprocess_block__social_post(&$variables) {
+  if (
+    \Drupal::routeMatch()->getRouteName() === 'social_album.post' &&
+    ($content = \Drupal::service('social_post.helper')->buildCurrentUserImage())
+  ) {
+    $variables['content'] = [
+      'form' => $variables['content'],
+      'current_user_image' => $content,
+    ];
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function social_album_preprocess_node(&$variables) {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $variables['node'];
+
+  if (
+    $node->bundle() === 'album' &&
+    $variables['view_mode'] === 'teaser' &&
+    views_get_view_result('albums', 'embed_album_cover', $node->id())
+  ) {
+    $variables['node_image'] = views_embed_view('albums', 'embed_album_cover', $node->id());
+    $variables['no_image'] = FALSE;
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function social_album_preprocess_views_view(&$variables) {
+  /** @var \Drupal\views\ViewExecutable $view */
+  $view = $variables['view'];
+
+  if ($view->id() === 'albums') {
+    $library = [
+      'embed_album_overview' => 'social_album/social_album_post',
+      'page_albums_overview' => 'social_album/social_album_teaser',
+    ];
+
+    if (isset($library[$view->current_display])) {
+      $variables['#attached']['library'][] = $library[$view->current_display];
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function social_album_preprocess_social_album_post(&$variables) {
+  $items = [];
+
+  foreach ($variables['items'] as $item) {
+    if (!isset($items[$pid = $item['pid']])) {
```

Comment:
Not sure if it is a good idea to create a new variable in the array brakets, also `$pid = $item['pid']`, and are you sure that the `$pid` is always not NULL?